File: JDK11/opentraceability/src/main/java/opentraceability/models/masterdata/TradeItem.java
package opentraceability.models.masterdata;

import java.util.List;
import java.util.ArrayList;

import org.apache.commons.codec.language.bm.Lang;
import org.json.JSONObject;
import opentraceability.interfaces.IMasterDataKDE;
import opentraceability.interfaces.IVocabularyElement;
import opentraceability.interfaces.VocabularyType;
import opentraceability.models.common.LanguageString;
import opentraceability.models.identifiers.GTIN;
import opentraceability.models.identifiers.PGLN;
import opentraceability.utility.attributes.OpenTraceabilityArrayAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;
import opentraceability.utility.attributes.OpenTraceabilityMasterDataAttribute;

public class TradeItem extends IVocabularyElement {
    public TradeItem()
    {
        epcisType = "urn:epcglobal:epcis:vtype:EPCClass";
        vocabularyType = VocabularyType.TradeItem;
    }
    @OpenTraceabilityJsonAttribute(name = "gtin")
    public GTIN gtin = null;

    @OpenTraceabilityArrayAttribute(itemType = LanguageString.class)
    @OpenTraceabilityJsonAttribute(name = "productName")
    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#descriptionShort")
    public List<LanguageString> shortDescription = new ArrayList<LanguageString>();

    @OpenTraceabilityJsonAttribute(name = "cbvmda:tradeItemConditionCode")
    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#tradeItemConditionCode")
    public String tradeItemConditionCode = null;
    @OpenTraceabilityJsonAttribute(name = "cbvmda:owning_party")
    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:owning_Party")
    public PGLN owningParty = null;
    @OpenTraceabilityJsonAttribute(name = "cbvmda:informationProvider")
    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#informationProvider")
    public PGLN informationProvider = null;
    @OpenTraceabilityArrayAttribute(itemType = String.class)
    @OpenTraceabilityJsonAttribute(name = "cbvmda:speciesForFisheryStatisticsPurposesName")
    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#speciesForFisheryStatisticsPurposesName")
    public List<String> fisherySpeciesScientificName = new ArrayList<String>();
    @OpenTraceabilityArrayAttribute(itemType = String.class)
    @OpenTraceabilityJsonAttribute(name = "cbvmda:speciesForFisheryStatisticsPurposesCode")
    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#speciesForFisheryStatisticsPurposesCode")
    public List<String> fisherySpeciesCode = new ArrayList<String>();
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/models/masterdata/tradeitem.kt
package opentraceability.models.masterdata

import opentraceability.interfaces.IMasterDataKDE
import opentraceability.interfaces.IVocabularyElement
import opentraceability.interfaces.VocabularyType
import opentraceability.models.common.LanguageString
import opentraceability.models.identifiers.GTIN
import opentraceability.models.identifiers.PGLN
import opentraceability.utility.attributes.OpenTraceabilityArrayAttribute
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute
import opentraceability.utility.attributes.OpenTraceabilityMasterDataAttribute
import java.util.*
import org.json.JSONObject

class TradeItem()  : IVocabularyElement {

    override var id: String? = null

    override var epcisType: String? = "urn:epcglobal:epcis:vtype:EPCClass"

    @OpenTraceabilityJsonAttribute("@type")
    override var jsonLDType: String? = "gs1:Product"

    override var vocabularyType: VocabularyType = VocabularyType.TradeItem

    override var context: JSONObject? = null

    @OpenTraceabilityJsonAttribute("gtin")
    var gtin: GTIN? = null

    @OpenTraceabilityJsonAttribute("productName")
    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#descriptionShort")
    var shortDescription: MutableList<LanguageString>? = mutableListOf()

    @OpenTraceabilityJsonAttribute("cbvmda:tradeItemConditionCode")
    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#tradeItemConditionCode")
    var tradeItemConditionCode: String? = null

    @OpenTraceabilityJsonAttribute("cbvmda:owning_party")
    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:owning_Party")
    var owningParty: PGLN? = null

    @OpenTraceabilityJsonAttribute("cbvmda:informationProvider")
    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#informationProvider")
    var informationProvider: PGLN? = null

    @OpenTraceabilityArrayAttribute
    @OpenTraceabilityJsonAttribute("cbvmda:speciesForFisheryStatisticsPurposesName")
    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#speciesForFisheryStatisticsPurposesName")
    var fisherySpeciesScientificName: MutableList<String>? = mutableListOf()

    @OpenTraceabilityArrayAttribute
    @OpenTraceabilityJsonAttribute("cbvmda:speciesForFisheryStatisticsPurposesCode")
    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#speciesForFisheryStatisticsPurposesCode")
    var fisherySpeciesCode: MutableList<String>? = mutableListOf()

    override var kdes: MutableList<IMasterDataKDE> = mutableListOf()
}
